app/models/book.rb

```
@@ -13,11 +13,33 @@
 #  chapters               :integer
 #  pages                  :integer
 #  narrative_structure_id :bigint
+#  protagonist_id         :bigint
 #
 class Book < ApplicationRecord
   belongs_to :writing_style
   belongs_to :perspective
   belongs_to :narrative_structure
+  belongs_to :protagonist, class_name: 'Character', optional: true
+
+  has_many :book_antagonists
```

RuboCop (1.69.1)

[convention] Specify a `:dependent` option.

(Rails/HasManyOrHasOneDependent)